docs/new-spec.md

```diff
@@ -0,0 +1,347 @@
+        sequence_number: &Uint256,
+        balance0: &Uint256,
+        balance1: &Uint256,
+        signature0: &Signature,
+        signature1: &Signature) -> impl Future<Item = (), Error = Error>;
+}
+```
+
+### PaymentManager
+
+This is what `guac_actix`'s `PaymentController` used to be, but is not tied to actix now. This more likely extracts its functionality in a generic layer. Exposing it through Actix actor will be trivial though, as explained later in [Actix adapter](#actix-adapter) section.
+
+```rust
+
+/// Tries to resemble most of the guac_actix's PaymentController stuff
+trait PaymentManager {
+    fn withdraw(&self, ...) -> impl Future<Item = (), Error = Error>;
+    fn make_payment(&self, counterparty: &Counterparty) -> impl Future<Item = (), Error = Error>;
+}
+
+struct GuacPaymentManager<T, C, S>
+where T: TransportProtocol, C: PaymentContract, S: Storage {
+    transport: TransportProtocol,
+    contract: PaymentContract,
+    storage: Storage,
+}
+
+struct GuacPaymentManager<T, C, S>
+    where T: TransportProtocol,
+          C: PaymentContract,
+          S: Storage {
+    fn new(t: T, c: C, s: S) -> Self {
+        Self {
+            transport: t,
+            contract: c,
+            storage: s,
+        }
+    }
+}
+
+impl<T, C, S> PaymentManager for GuacPaymentManager<T, C, S> {
+    fn withdraw(&self, ...) -> impl Future<Item = (), Error = Error> {
+        /// TODO
+    }
+    fn make_payment(&self, counterparty: &Counterparty, amount: Uint256) -> impl Future<Item = (), Error = Error> {
+        Box::new(self.storage.get_channel(counterparty.eth_address).and_then(
+            move |mut channel_manager| {
+                channel_manager.pay_counterparty(Uint256(amount));
+            }
+        ));
+    }
+
+    fn tick(&self) -> impl Future<Item = (), Error = Error> {
+        Box::new(self.storage.get_all_counterparties().and_then(|keys| {
+            for i in keys {
+                self.storage
+                    .get_channel(counterparty.address.clone())
+                    .and_then(move |mut channel_manager| {
+
+                        let action = channel_manager.tick();
+
+                        match action {
+                            // Use self.transport for HTTP requests between parties
+                            // Use self.contract for Contract requests
+                            // Use self.storage for accessing counterparties
+                        }
+                    });
+            }
+        });
+    }
+}
+
+```
+
+### Channel manager
+
+Channel manager is code is mostly left in place, just removed parts of code where it does not match the functionality in Guac contract.
+
+```rust
+enum ChannelManager {
+    // This code is mostly left intact
+    New,
+    Proposed {
+        state: Channel,
+        accepted: bool,
+    },
+    Open {
+        state: CombinedState,
+    },
+    Closed {
+        state: CombinedState,
+    }
+    // ...
+}
+
+impl ChannelManager {
```

Comment:
isn't this internal and not documentation material then? Or at least shouldn't we be talking about how it works and exchanges info.